doc/source/data/key-concepts.rst

```diff
@@ -143,10 +188,67 @@ Datasets uses the Ray object store to store data blocks, which means it inherits
 
 **Load Balancing**: Datasets uses Ray scheduling hints to spread read tasks out across the cluster to balance memory usage.
 
```

Comment:
It'd be great to have a section about Fault Tolerance. But we may separate that out from this PR if swang@ is already working on that.

Reply:
Agreed, let's wait for @stephanie-wang's PR for that.